packages/bitcore-node/src/providers/chain-state/evm/api/ecsp.ts

```diff
@@ -114,103 +118,196 @@ export class BaseEVMExternalStateProvider extends InternalStateProvider implemen
-          tx.receipt = receipt;
-          tx.fee = fee;
-        }
-        const tansformedTx = this.transformTransactionData({ tx, network, chain });
-        const convertedTx = EVMTransactionStorage._apiTransform(tansformedTx, { object: true }) as EVMTransactionJSON;
-        return { ...convertedTx, confirmations };
-      } else {
-        return undefined;
-      }
+      return await this._getTransaction({ chain, network, txId, tipHeight, web3 });
     } catch (err) {
-      console.error(err);
+      logger.error('Error getting transations from historical node %o', err);
     }
     return undefined;
   }
 
-  async getWalletBalanceAtTime(_params: GetWalletBalanceAtTimeParams): Promise<{ confirmed: number; unconfirmed: number; balance: number }> {
-    throw new Error('Method not implemented.');
+  async _getTransaction({ chain, network, txId, tipHeight, web3 }) {
+    const tx: any = await web3.eth.getTransaction(txId);
+    if (tx) {
+      let confirmations = 0;
+      if (tx.blockNumber && tx.blockNumber >= 0) {
+        confirmations = tipHeight - tx.blockNumber + 1;
+      }
+      tx.blockTime = await this.getBlockTimestamp(tx.blockHash, { network });
+      const receipt = await web3.eth.getTransactionReceipt(txId);
+      if (receipt) {
+        const fee = receipt.gasUsed * parseInt(tx.gasPrice);
+        tx.receipt = receipt;
+        tx.fee = fee;
+      }
+      const tansformedTx = ECSP.transformTransactionData({ tx, network, chain });
+      const convertedTx = EVMTransactionStorage._apiTransform(tansformedTx, { object: true }) as EVMTransactionJSON;
+      return { ...convertedTx, confirmations };
+    }
+    return undefined;
   }
 
-  async streamBlocks(_params: StreamBlocksParams) {
-    throw new Error('Method not implemented.');
+  // stream all transactions from a desired block
+  async streamTransactions(params: StreamTransactionsParams) {
+    const { chain, network, req, res, args } = params;
+    let { blockHash, blockHeight } = args;
+    try {
+      if (blockHeight !== undefined) {
+        blockHeight = Number(blockHeight);
+      }
+      if (blockHash !== undefined) {
+        blockHash = blockHash;
+      }
+      // get block tip for confirmations
+      const tip = await this.getLocalTip(params);
+      const tipHeight = tip ? tip.height : 0;
+      const { web3 } = await this.getWeb3(network, { type: 'historical' });
+      // get block with desired transactions
+      const block = await web3.eth.getBlock(blockHash || blockHeight);
+      const getTransaction = (async (txId) => {
+        return await this._getTransaction({ chain, network, txId, tipHeight, web3 });
+      }).bind(this);
+      const txStream = new NodeQueryStream(block?.transactions || [], getTransaction, args);
+      // stream results into response
+      await NodeQueryStream.onStream(txStream, req!, res!);
+    } catch (err) {
+      logger.error('Error streaming transactions from historical node %o', err);
+      throw err;
```

Comment:
I'm getting an uncaught from a bum connection. It logs above, then this throw is uncaught upstream. I think you need an await on `ChainStateProvider.streamAddressTransactions` in address.ts > streamCoins()